src/lab3/types/Transformable.ts

```diff
@@ -0,0 +1,17 @@
+import { Traceable } from '../../lab1/types/Traceable';
+
+export interface Rotatable {
+  rotate(angleX: number, angleY: number, angleZ: number): void;
+}
+
+export interface Translatable {
+  translate(x: number, y: number, z: number): void;
+}
+
+export interface Scalable {
+  scale(x: number, y: number, z: number): void;
+}
+
+export interface Transformable extends Rotatable, Translatable, Scalable {}
```

Comment:
Either `Traceable` should extend this `Transformable` interface or just make all classes with `implements Traceable, Transformable`. `SceneObject` interface's name doesn't clearly describe its implementation abilities

Reply:
Having Traceable extend Transformable would be a violation of interface segregation principle. SceneObject here is just a shortcut to always writing `Traceable, Transformable`, but I can rename it to something like TraceableTransformable to be clear